include/ppx/fs.h

```diff
@@ -143,6 +143,13 @@ std::filesystem::path GetInternalDataPath();
 std::filesystem::path GetExternalDataPath();
 #endif
 
+// Returns the default output directory where test artifacts will be written.
+std::filesystem::path GetDefaultOutputDirectory();
+
+// Constructs the full path from a possibly partial path
+// If `regexToReplace` is specified, replace all instances of that symbol in the filename with `replaceString`
+std::filesystem::path GetFullPath(const std::filesystem::path& partialPath, const std::filesystem::path& defaultFolder, const std::string& regexToReplace = "", const std::string& replaceString = "");
```

Comment:
Since we don't have an existing use case for passing defaultPath as anything other than GetDefaultOutputDirectory(), should we just call that inside the function instead of passing as a parameter?

Reply:
I think it makes it easier to unit test if this is separate so I'd prefer to keep it like this.